src/implementations/wrap/wrap.ts

```diff
@@ -71,14 +66,39 @@ export const digestDocument = async (
 
 export const appendProofToDocuments = async (
   intermediateDir: string,
-  digestedDocumentDir: string,
-  hashMap: Record<string, { sibling: string; parent: string }>
+  digestedDocumentPath: string,
+  hashMap: Record<string, { sibling: string; parent: string }>,
+  outputPathType: "file" | "directory" | "stdOut"
 ): Promise<string> => {
   const documentFileNames = await documentsInDirectory(intermediateDir);
   let merkleRoot = "";
-  documentFileNames.forEach(file => {
-    const document = readDocumentFile(file);
 
+  if (outputPathType == "stdOut") {
```

Comment:
the logic to compute merkle root and other should be the same whatever the input. only writing should be different => there is a lot of redundancy to remove here

Reply:
fixed